### PR TITLE
Force Inductor output code to be dumped even if it fails to compile

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #127700

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/graph.py [hunk 1]
```diff
@@ -1688,8 +1688,25 @@ def compile_to_module(self):
...
+                # Just omit the filename, I still want the code though!
+                payload_fn=lambda: code,
+            )
+            raise
+        else:
+            trace_structured(
+                "inductor_output_code",
+                lambda: {"filename": path},
+                payload_fn=lambda: code,
+            )
```
Comment (oulgen) [inline torch/_inductor/graph.py:1708]: can you use try-finally here so you don't duplicate the code? Just make the path be None above the try

**Review by oulgen — APPROVED** [review]

File: torch/_inductor/graph.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/graph.py:1708]: If we hate the ugliness here, I would concretely propose we just drop the filename from the logs lol

**Pull request closed without merging**